### ISSUE TITLE
Please check FQDN check string

### ISSUE THREAD

**Issue by Mirocow** [certbot/certbot#2687]
In this line https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/le_util.py#L341

this domain ws.dev.somedomain.com don`t check by regular string

```
^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,63}$
```

but such it work

```
^((?!-)[[A-Za-z0-9-]{1,63}(?<!-)\.)+[A-Za-z]{2,6}$
```

**Comment by joohoi** [comment]
What version of python are you running? It should match:

```
$ python -c 'import re;print(re.compile("^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,63}$").match("ws.dev.somedomain.com").string)'
ws.dev.somedomain.com
```

**Comment by Mirocow** [comment]
2.7

**Comment by joohoi** [comment]
That's strange, it matches fine in 2.7 for me too. Please provide steps to reproduce.

**Comment by ohemorange** [comment]
Seems that this was fixed in #3061.
